### PR TITLE
Play button and resume

### PR DESCRIPTION
**Pull request by MrTimscampi**
(I forgot to open a PR for this one and the branch has been sitting there for a few days :laughing: )

This allows resuming if an item is already in progress, and adds a "Play from the beginning" entry to the item menu, which shows up when you can resume an item.

### CODE REVIEW

File: pages/genre/_itemId/index.vue [hunk 1]
```diff
@@ -3,17 +3,7 @@
...
-          class="play-button ml-auto mr-2"
-          color="primary"
-          min-width="8em"
-          depressed
-          rounded
-          @click="play({ items: [genre] })"
-        >
-          {{ $t('play') }}
-        </v-btn>
+        <v-btn v-if="loaded" :item="genre" />
```
Comment (Camc314) [inline pages/genre/_itemId/index.vue:6]:
```suggestion
        <play-button v-if="loaded" :item="genre" />

**Review by Camc314 — COMMENTED** [review]
I seem unable to resume anything. The video interface just ends up as a black screen. I have checked the source url and it works.

**Comment by MrTimscampi** [comment]
> I seem unable to resume anything. The video interface just ends up as a black screen. I have checked the source url and it works.

This is linked to streams being unseekable on master, I believe. Since you can't seek in the file, it'll start from the beginning even with the proper position passed to Shaka.

**Comment by Camc314** [comment]
> > I seem unable to resume anything. The video interface just ends up as a black screen. I have checked the source url and it works.
> 
> This is linked to streams being unseekable on master, I believe. Since you can't seek in the file, it'll start from the beginning even with the proper position passed to Shaka.

Ah, that would make sense, I'll roll back to RC2 and see whether it works with that.

**Comment by MrTimscampi** [comment]
> I'll roll back to RC2 and see whether it works with that.

The webm issue is on our end, since web doesn't have the issue :p It's likely due to some change in the profile. Ours is very different compared to jf-web, which it shouldn't be :/

File: components/Players/PlayerManager.vue [hunk 2]
```diff
@@ -239,28 +238,30 @@ export default Vue.extend({
           this.setLastProgressUpdate({ progress: new Date().getTime() });
           break;
         case 'playbackManager/SET_CURRENT_TIME': {
-          const now = new Date().getTime();
+          if (state.playbackManager.status === PlaybackStatus.playing) {
```
Comment (ferferga) [inline components/Players/PlayerManager.vue:241]: Why?

File: components/Players/PlayerManager.vue [hunk 3]
```diff
@@ -239,28 +238,30 @@ export default Vue.extend({
...
+                    PositionTicks: Math.round(
+                      this.msToTicks(state.playbackManager.currentTime * 1000)
+                    )
+                  }
+                },
+                { progress: false }
+              );
 
-            this.setLastProgressUpdate({ progress: new Date().getTime() });
+              this.setLastProgressUpdate({ progress: new Date().getTime() });
```
Comment (ferferga) [inline components/Players/PlayerManager.vue:263]: Can't we use ``now`` here?

Comment (ferferga) [reply components/Players/PlayerManager.vue:263]: @MrTimscampi I don't understand why it's outside the scope. You're already declaring ``const now`` in line 242 (as I mentioned above), why not use that instead?

Comment (MrTimscampi) [reply components/Players/PlayerManager.vue:263]: This is all untouched code though, it just has an indentation change :)

**Review by ferferga — APPROVED** [review]

**Pull request merged**